### ISSUE TITLE
hostsupdater needs root/sudo access

### ISSUE THREAD

**Issue by maweki** [chad-thompson/vagrantpress#100]
since vagrantpress uses hostsupdater, the users need root/sudo access on their machines and the password needs to be entered on startup and halting.

I think, that this is a regression.

**Comment by akeif** [comment]
@maweki I don't like it either. This has been opened for a while. Did you find a solution on your own?

I see two solutions:

1) Add instruction to make hostupdater sudoless.
2) Make hostupdater optionnal

**Comment by maweki** [comment]
@akeif yeah. My solution was forking the project, removing hostsupdater, and cherry-picking the patches I like.

Not the best, I know.

**Comment by akeif** [comment]
@maweki Not the best but it works ;)

I think I can tackle this issue. I'll take a look and let you know.

**Comment by akeif** [comment]
All right @maweki. I got the solution and it works on my workstation. I don't think there's a way to update change Vagrantpress to make `vagrant up|down` password less for everybody.

Here's how you do it: `sudo visudo`

then add

    # Allow passwordless startup of Vagrant with vagrant-hostsupdater.
    Cmnd_Alias VAGRANT_HOSTS_ADD = /bin/sh -c echo "*" >> /etc/hosts
    Cmnd_Alias VAGRANT_HOSTS_REMOVE = /usr/bin/env sed -i -e /*/ d /etc/hosts
    %admin ALL=(root) NOPASSWD: VAGRANT_HOSTS_ADD, VAGRANT_HOSTS_REMOVE

if you're on Ubuntu 14 or newer the second line is

    Cmnd_Alias VAGRANT_HOSTS_REMOVE = /bin/sed -i -e /*/ d /etc/hosts

make sure you change `%admin` for your user name or a group you're in.

[Official documentation](https://github.com/cogitatio/vagrant-hostsupdater#passwordless-sudo) for this pop on hitting CTRL+C at the password prompt for hostmanager during `vagrant up|down`.

    ==> default: Waiting for cleanup before exiting...
    ==> default: [vagrant-hostsupdater] Failed to add hosts, could not use sudo
    ==> default: [vagrant-hostsupdater] Consider adding the following to your sudoers file:
    ==> default: [vagrant-hostsupdater]   https://github.com/cogitatio/vagrant-hostsupdater#passwordless-sudo
    Vagrant exited after cleanup due to external interrupt.

**Comment by akeif** [comment]
I added this to the [project wiki](https://github.com/vagrantpress/vagrantpress/wiki/HOWTO:-Password-less-vagrant-up%7Cdown).

Look like it's the best we can do.

**Comment by akeif** [comment]
@battlemidget I think we can close this ticket, though, you might want to wait for @maweki 's feedback.

**Comment by battlemidget** [comment]
@akeif thanks ill wait a day for any additional feedback then close this out

**Comment by maweki** [comment]
I don't like any solution that requires elevated privileges in any way, but I guess, considering the way dns-resolution works, there might be no better one. I'll have to ask my coworkers if they are more comfortable with localhost:8080 than an injected hostname.

But thanks for the effort :)

**Comment by akeif** [comment]
Hum. I might have a solution for you then. I'll test it out tomorrow and
let you know.

**Comment by akeif** [comment]
@maweki If you manually add the entries in your `hosts` file, the host manager won't ask for admin rights.

    ==> default: [vagrant-hostsupdater] Checking for host entries
    ==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 vagrantpress.dev
    ==> default: [vagrant-hostsupdater]   found entry for: 192.168.50.4 www.vagrantpress.dev

**Comment by akeif** [comment]
@battlemidget I added both solutions to the [wiki](https://github.com/vagrantpress/vagrantpress/wiki/HOWTO:-Password-less-vagrant-up|down).

I think it would be a good idea to add this link to the project's README.

**Comment by mikestankavich** [comment]
I know that it doesn't directly address the need for sudo, but the PR that I just created https://github.com/vagrantpress/vagrantpress/pull/126 does add support for the vagrant-hostmanager plugin, which at least gives more flexibility on updating /etc/hosts.
